presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1141,39 +1146,31 @@ public void alterTable(HiveIdentity identity, String databaseName, String tableN
     }
 
     @Override
-    public Optional<List<String>> getPartitionNames(HiveIdentity identity, String databaseName, String tableName)
+    public Optional<List<String>> getPartitionNamesByFilter(HiveIdentity identity, String databaseName, String tableName, List<String> columnNames, TupleDomain<String> partitionKeysFilter)
     {
-        try {
-            return retry()
-                    .stopOn(NoSuchObjectException.class)
-                    .stopOnIllegalExceptions()
-                    .run("getPartitionNames", stats.getGetPartitionNames().wrap(() -> {
-                        try (ThriftMetastoreClient client = createMetastoreClient(identity)) {
-                            return Optional.of(client.getPartitionNames(databaseName, tableName));
-                        }
-                    }));
-        }
-        catch (NoSuchObjectException e) {
-            return Optional.empty();
-        }
-        catch (TException e) {
-            throw new PrestoException(HIVE_METASTORE_ERROR, e);
-        }
-        catch (Exception e) {
-            throw propagate(e);
+        Optional<List<String>> parts = partitionKeyFilterToStringList(columnNames, partitionKeysFilter, assumeCanonicalPartitionKeys);
+        Preconditions.checkArgument(!columnNames.isEmpty() || partitionKeysFilter.isAll(), "must pass in all columnNames or the filter must be all");
+        if (parts.isEmpty()) {
+            return Optional.of(ImmutableList.of());
         }
-    }
 
-    @Override
-    public Optional<List<String>> getPartitionNamesByParts(HiveIdentity identity, String databaseName, String tableName, List<String> parts)
-    {
         try {
+            if (partitionKeysFilter.isAll()) {
+                return retry()
+                        .stopOn(NoSuchObjectException.class)
+                        .stopOnIllegalExceptions()
+                        .run("getPartitionNames", stats.getGetPartitionNames().wrap(() -> {
+                            try (ThriftMetastoreClient client = createMetastoreClient(identity)) {
+                                return Optional.of(client.getPartitionNames(databaseName, tableName));
+                            }
+                        }));
+            }
             return retry()
                     .stopOn(NoSuchObjectException.class)
                     .stopOnIllegalExceptions()
                     .run("getPartitionNamesByParts", stats.getGetPartitionNamesByParts().wrap(() -> {
                         try (ThriftMetastoreClient client = createMetastoreClient(identity)) {
-                            return Optional.of(client.getPartitionNamesFiltered(databaseName, tableName, parts));
+                            return Optional.of(client.getPartitionNamesFiltered(databaseName, tableName, parts.get()));
```

Comment:
the `"getPartitionNamesByParts", stats.getGetPartitionNamesByParts()...` above should be renamed too.

Reply:
nvrm, i see this is pre-existing... still worth updating